modelserver-markers-theia/README.md

```diff
@@ -0,0 +1,47 @@
+# Markers management for Model Server validation Diagnostics
+
+## Use case
+
+The model server client API provides validation methods which manipulate JSON objects which can be cast as `Diagnostic` objects (`@eclipse-emfcloud/modelserver-theia/lib/browser/diagnostic`).
+
+```typescript
+export interface ModelServerClient extends JsonRpcServer<ModelServerFrontendClient> {
+    validation(modelUri: string): Promise<Response<string>>;
+
+    // WebSocket connection
+    subscribeWithValidation(modelUri: string): void;
+}
+```
+
+Users may want to log these validation diagnostics in Theia's Problems view (provided by `@theia/markers`).
+This module provides the necessary tools to achieve this purpose with minimalist code.
+
+## Typescript API Examples
+
+After calling the validation, you can create markers as simply as :
+
+```typescript
+
+@inject(MessageService) protected readonly messageService: MessageService;
+@inject(DiagnosticManager) protected readonly diagnosticManager: DiagnosticManager;
+
+const modelURI : URI;
+// perform validation
+this.modelServerClient.validation(modelURI.toString())
+    .then((response: Response<any>) => {
+        // get resulting diagnostic
+        const diagnostic = response.body as Diagnostic;
+        // print markers in Problems view
+        diagnosticManager.setDiagnostic(modelURI, diagnostic);
+        // display the validation status
+        this.messageService.info(`Validation finished with level ${Diagnostic.getSeverityLabel(diagnostic)}.`)
+    });
+            
+```
+
+## Selecting the model element from the marker
+
+The marker in the problems view can be used to navigate to the precise element in error.
+As this navigation depends on your model editor implementation, you must implement the `OpenHandler` interface if you want this functionality (see `@theia/core/lib/browser/opener-service.ts`).
+In the opener options, you will find a `selection` entry, which value can be cast as a `ModelElementRange`, containing fake start and end values, but the correct model element's URI fragment (identifier) at the `uriFragment` key.
+This is up to your widget implementation to select the correct element corresponding to this URI fragment.
```

Comment:
`This is up to your... to select...` sounds weird to me but I am not native speaker so it might be correct. Maybe `It is up to your... to select...`.

Reply:
I ain't either... I think both may be corrects, but "It is up to" is surely more frequent.
I'm positively sure that "that is up to" would be incorrect here because it is used only for decision (whereas we meant responsibility here).
The reference I found seems to suggest that "this is up to" also means decision
https://www.dict.cc/englisch-deutsch/This+is+up+to+you.html

Reply:
Right, then let's just leave it and not over-complicate things!